tests/timelock-controller.tests.ts

```diff
@@ -0,0 +1,172 @@
+        const signers = await getSigners()
+        const bob = await getRandomSigner(signers[1], one.muln(10000))
+        let contract = await setupContract('my_timelock_controller', 'new', 0, [bob.address], [bob.address]);
+
+        return { contract, bob }
+    }
+
+    it('TIMELOCK CONTROLLER - can schedule', async () => {
+        const {
+            contract,
+            bob,
+        } = await setup()
+
+        // Arrange - Prepare data for schedule
+        let transaction = {
+            callee: contract.contract.address,
+            selector: [0, 0, 0, 0],
+            input: [],
+            transferred_value: 0,
+            gas_limit: 0,
+        }
+        let salt = bnArg(0);
+
+
+        // Act - Bob scheduled the transaction
+        // @ts-ignore
+        let id = (await contract.query.hashOperation(transaction, null, salt)).output!;
+        await expect(contract.query.isOperationPending(id)).to.have.output(false)
+        // @ts-ignore
+        await expect(fromSigner(contract.contract, bob.address).tx.schedule(transaction, null, salt, 0)).to.eventually.be.fulfilled
+
+        // Assert - Operation must be scheduled, it is in Pending state an in Ready state(because min delay si zero)
+        await expect(contract.query.isOperationPending(id)).to.have.output(true)
+        await expect(contract.query.isOperationReady(id)).to.have.output(true)
+        await expect(contract.query.isOperationDone(id)).to.have.output(false)
+    })
+
+    it('TIMELOCK CONTROLLER - schedule and execute without input data `TimelockController,get_min_delay`', async () => {
+        const {
+            contract,
+            bob,
+        } = await setup()
+
+        // Arrange - Prepare data for execute `get_min_delay`
+        const message = contract.contract.abi.messages.find(message => message.identifier === "TimelockController,get_min_delay")!;
+        let transaction = {
+            callee: contract.contract.address,
+            selector: message.selector,
+            input: [],
+            transferred_value: 0,
+            gas_limit: 0,
+        }
+        let salt = bnArg(0);
+
+        // Act - Bob scheduled the transaction
+        // @ts-ignore
+        let id = (await contract.query.hashOperation(transaction, null, salt)).output!;
+        // @ts-ignore
+        await expect(fromSigner(contract.contract, bob.address).tx.schedule(transaction, null, salt, 0)).to.eventually.be.fulfilled
+
+        // Assert - Transaction must be updated and now the state is Done
+        await expect(contract.query.isOperationDone(id)).to.have.output(false)
+        // @ts-ignore
+        await expect(fromSigner(contract.contract, bob.address).tx.execute(transaction, null, salt)).to.eventually.be.fulfilled
+        await expect(contract.query.isOperationDone(id)).to.have.output(true)
+    })
+
+    it('TIMELOCK CONTROLLER - schedule and execute by passing value into `TimelockController,update_delay`, and update', async () => {
```

Comment:
```suggestion
    it('TIMELOCK CONTROLLER - schedule and execute by passing value into `TimelockController.update_delay`, and update', async () => {
```

Reply:
The idea is to use a comma) Because it is the identifier of the method